pkg/dbmate/db.go

```diff
@@ -423,7 +428,7 @@ func (db *DB) printVerbose(result sql.Result) {
 }
 
 func (db *DB) readMigrationsDir(dir string) ([]fs.DirEntry, error) {
-	path := filepath.Clean(dir)
+	path := path.Clean(dir)
```

Comment:
I'm a bit confused as to how the use of `filepath.Clean` results in the wrong behavior, as the entire sales pitch for `filepath` is that the utilities are supposed to Do The Right Thing&trade; according to the platform being used:

https://pkg.go.dev/path/filepath

> The filepath package uses either forward slashes or backslashes, depending on the operating system. To process paths such as URLs that always use forward slashes regardless of the operating system, see the [path](https://pkg.go.dev/path) package.

As `readMigrationsDir` is operating on filesystem paths and not URLs, normalizing forward slashes to backslashes when executing on Windows seems like the exact behavior that would be required here.

Could you add a automated test that, without your change, passes on Linux but fails on Windows, to clearly illustrate the issue, and hopefully prevent regressions in the future, as I can absolutely see someone reverting this change unintentionally, thinking that the use of `path.Clean` is wrong for cross-platform use and replace it with `filepath.Clean` again.

I'm mostly concerned that this is not a defect in dbmate, and that the use of `filepath` utils is correct cross-platform, and that you are actually running into a bug in go's [embed](https://pkg.go.dev/embed), and specifically `embed.FS` on Windows not using the correct path separator:

> The path separator is a forward slash, even on Windows systems.

This behavior is WRONG, and is a defect of the `embed.FS` implementation, because it's supposed to be transparent, otherwise now developers need to have code that uses `filepath` functions when operating on real filesystems and the `path` functions when operating on an `embed.FS` which ruins the abstraction.

Sadly, it seems like `io.FS` and `embed.FS` are sticking to their guns and insisting on this behavior: https://github.com/golang/go/issues/44305

> You should use path.Join, not filepath.Join for embed filenames.

And, we have this: https://pkg.go.dev/io/fs#ValidPath

> Note that paths are slash-separated on all systems, even Windows. Paths containing other characters such as backslash and colon are accepted as valid, but those characters must never be interpreted by an [FS](https://pkg.go.dev/io/fs#FS) implementation as path element separators.

So, yeah, it looks like your change is correct and `path/filepath` should not be used when using the `FS` filesystem layer abstraction.  Confusing, I know ...

Anyway, could you add a test case that fails on Windows without the change in this PR?  I think that would help ensure we don't regress this by accident in the future through some well-meaning change.

Reply:
Very good review @dossy , thanks!

Regarding the test you asked, I may have not correctly understood you but the `TestFindMigrationsFS` test in `pkg/dbmate/db_test.go` already fails on Windows without merging this PR.

Reply:
> Regarding the test you asked, I may have not correctly understood you but the `TestFindMigrationsFS` test in `pkg/dbmate/db_test.go` already fails on Windows without merging this PR.

Perfect!  Can you tell I've never run dbmate or its test suite on Windows, myself?  :-)

I tried to build dbmate on Windows using Cygwin and MSYS2/ucrt64, and I get this error trying to build:

```
$ make build
rm -rf dist
go build -o dist/dbmate.exe -tags sqlite_omit_load_extension,sqlite_json -ldflags '-s'  .
# runtime/cgo
gcc_libinit_windows.c: In function ‘_cgo_beginthread’:
gcc_libinit_windows.c:143:27: error: implicit declaration of function ‘_beginthread’; did you mean ‘_cgo_beginthread’? [-Werror=implicit-function-declaration]
  143 |                 thandle = _beginthread(func, 0, arg);
      |                           ^~~~~~~~~~~~
      |                           _cgo_beginthread
cc1: all warnings being treated as errors
make: *** [Makefile:34: build] Error 1
```

How were you able to build dbmate on Windows?

Reply:
Answering my own question, I installed the `mingw-w64-ucrt-x86_64-toolchain` package in MSYS2, and then I was able to build dbmate successfully.  Now, to try and run some of the test suite without having Docker: my Windows environment is a VMware VM ... I don't want to even try running Docker inside a Windows VMware VM ...